keyboards/hhkb/keymaps/schaeferdev/keymap.c

```diff
@@ -0,0 +1,96 @@
+                 | **** | **** | ******************** | **** | **** |
+                 |------+------+----------------------+------+------+
+     */
+
+    [HHKB] = LAYOUT(
+        KC_PWR, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_DEL, KC_DEL,
+        KC_CAPS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PSCR, KC_SLCK, KC_PAUS, KC_UP, KC_TRNS, KC_BSPC,
+        KC_TRNS, KC_VOLD, KC_VOLU, KC_MUTE, KC_TRNS, KC_TRNS, KC_PAST, KC_PSLS, KC_HOME, KC_PGUP, KC_LEFT, KC_RGHT, KC_PENT,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PPLS, KC_PMNS, KC_END, KC_PGDN, KC_DOWN, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+
+    /* Layer MOUSE: MOUSE mode (MOUSE Fn)
+      |------+-----+-----+-----+----+----+--------+----------------+--------------+-----------+-------------+-----+-------+-------+-----|
+      | `~   |ACC1 |ACC2 |ACC3 |    |    |        |                |              |           |             |     |       | DEL   | DEL |
+      |------+-----+-----+-----+----+----+--------+----------------+--------------+-----------+-------------+-----+-------+-------+-----|
+      |      |MBTN2|     | up  |    | PgU|        |                |              |           |             |     |       |       |     |
+      |------+-----+-----+-----+----+----+--------+----------------+--------------+-----------+-------------+-----+-------+-------+-----|
+      |      |MBTN1|right|down |left| PgD|  <-    |  down arrow    |   up arrow   |  ->       |             |     |       |       |     |
+      |------+-----+-----+-----+----+----+--------+----------------+--------------+-----------+-------------+-----+-------+-------+-----|
+      |      |MBTN3|     |     |    |    |        |     ~ (N)      |              |           |             |     |       |       |     |
+      |------+-----+-----+-----+----+----+--------+----------------+--------------+-----------+-------------+-----+-------+-------+-----|
+                 |------+------+----------------------+------+------+
+                 | **** | **** | ******************** | **** | **** |
+                 |------+------+----------------------+------+------+
+     */
+
+    [MOVEMENT] = LAYOUT(
+        KC_GRV, KC_ACL0, KC_ACL1, KC_ACL2, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_BTN2, KC_TRNS, KC_MS_U, KC_TRNS, KC_PGUP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_BTN1, KC_MS_L, KC_MS_D, KC_MS_R, KC_PGDN, KC_LEFT, KC_DOWN, KC_UP, KC_RIGHT, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_BTN3, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TILD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)};
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt)
```

Comment:
Could you remove the `action_get_macro` function?  It's been deprecated in favor of `process_record_user`.

Reply:
Thanks for your suggestions and review!

Even after looking through documentation, I am not quite sure how replacing `action_get_macro` should work.

I also wasn't able to find any hhkb layout that doesnt implement the `action_get_macro` function. Could you lead me into the right direction?

Reply:
Well, for you, it's not doing anything, actually. 

So, you could remove it without causing any issues.

As for replacing it, you'd use `process_record_user` which includes much of the same functionality, but also allows for control over built in keycodes, too. 

https://docs.qmk.fm/#/feature_macros

Reply:
ah okay, I read over https://docs.qmk.fm/#/feature_macros?id=the-old-way-macro-action_get_macro and it was not quite clear to me how my `case 0:` would be represented using the 'new way of doing it'.

I will remove this function entirely, thanks again!

Reply:
Ah, yeah, I can understand that.  But for the `process_record`, you're actually adding new keycodes that can be used.  Or you can modify the functionality of existing keycodes. It's very neat. :)

Reply:
`action_get_macro` is now removed entirely for my layout.

I see, so basically I will only need cases that cover my defined macros, everything else will simply`return true` and I should be perfectly fine right?